src/chunk_dispatch_state.c

```diff
@@ -100,12 +99,15 @@ chunk_dispatch_exec(CustomScanState *node)
 				 * (See:
 				 * https://github.com/postgres/postgres/commit/ee0a1fc84eb29c916687dc5bd26909401d3aa8cd).
 				 */
+				MemoryContextSwitchTo(estate->es_query_cxt);
```

Comment:
I think it's better to use `old = MemoryContextSwitchTo(estate->es_query_cxt); ... MemoryContextSwitchTo(old);` idiom instead of assuming the old context was `GetPerTupleMemoryContext(estate)`

Reply:
i disagree with using old here, we switch to per tuple context explicitly on line 69 above in the same function, if i were to follow your advice i would have to introduce 2nd memorycontext variable which would not improve the code imo

Reply:
Ah you are right. Did not see the code above